docs/develop/financial-accounting.md

```diff
@@ -0,0 +1,187 @@
+
+From your perspective, your liability to the bank increased by $100 while your assets also increased
+by $100. From the bank's perspective, their assets (the loan to you) increased by $100 while their
+liabilities (the money in your bank account) also increased by $100.
+
+Double-entry bookkeeping ensures that funds are always accounted for. Money never just appears.
+**Funds always go from somewhere to somewhere.**
+
+## Keeping Accounts in Balance
+
+Now we understand that there are different types of accounts and every transaction will be recorded
+in two (or more) accounts -- but which accounts?
+
+The [Fundamental Accounting Equation](https://en.wikipedia.org/wiki/Accounting_equation) stipulates
+that:
+
+**Assets = Liabilities + Equity**
+
+Using our loan example, it's no accident that the loan increases assets and liabilities at the same
+time. Assets and liabilities are on the opposite sides of the equation, and both sides must be
+exactly equal. Loans increase assets and liabilities equally.
+
+Here are some other types of transactions that would affect assets, liabilities, and equity, while
+maintaining this balance:
+
+- If you withdraw $100 in cash from your bank account, your total assets stay the same. Your bank
+  account balance (an asset) would decrease while your physical cash (another asset) would increase.
+- From the perspective of the bank, you withdrawing $100 in cash decreases their assets in the form
+  of the cash they give you, while also decreasing their liabilities because your bank balance
+  decreases as well.
+- If a shareholder invests $1000 in the bank, that increases both the bank's assets and equity.
+
+Assets, liabilities, and equity are types of accounts on the balance sheet, meaning they represent a
+point in time. The other two main categories, revenue and expenses, are income statement accounts,
+meaning they represent flows of money in and out.
+
+Revenue and expenses impact the balance sheet over time. The expanded accounting equation can be
+written as:
+
+**Assets = Liabilities + Contributed Capital + Revenue − Expenses − Dividends**
+
+You don't need to memorize these equations (unless you're training as an accountant!). However, it
+is useful to understand that those main account types lie on different sides of this equation.
+
+## Debits and Credits vs Signed Integers
+
+Instead of using a positive or negative integer to track a balance, TigerBeetle and double-entry
+bookkeeping systems use **debits and credits**.
+
+The two entries that give "double-entry bookkeeping" its name are the debit and the credit: every
+transaction has at least one debit and at least one credit. (Note that for efficiency's sake,
+TigerBeetle `Transfer`s consist of exactly one debit and one credit. These can be composed into more
+complex [multi-debit, multi-credit transfers](./recipes/multi-debit-credit-transfers.md).) Which
+entry is the debit and which is the credit?
```

Comment:
Would be great to have some nice Aha moment here and say, "well the answer is easy once you understand that accounting is in fact... a type system! ... an account increases on the debit/credit side according to its type."